tests/sql/test_sql_show_tables.py

```diff
@@ -54,6 +54,7 @@ def test_show_tables_all(sess):
     assert actual == expect
 
 
+@pytest.skip(reason="`SHOW TABLES LIKE` is not properly supported yet")
```

Comment:
Maybe we can have a tracking issue? This way we can list MemoryCatalog support there.

Reply:
I'll make an issue. 

Note: the reason why I removed it is because it was a hack before that used `LIKE` as a "has prefix" instead of a proper `LIKE`. We might want to think more deeply about `list_tables` and `list_namespaces` works because it's not a great abstraction right now. Maybe the pattern could be treated as some sort of pushdown instead?

Reply:
The idea was for this parity `list_tables(pattern: str)`  <-> `show tables like <pattern>` and we can discuss on the existing issue. I know this has kept coming up, so hopefully any feedback can be recorded in the dedicated issue and referenced. Yes to the pushdown, [that's what I was referring to here](https://github.com/Eventual-Inc/Daft/pull/4394#discussion_r2105494502).

* #4007

Reply:
#4461

Reply:
But yeah disabling for now since it's not a major features so I would prefer it to not work rather than to produce unexpected results.

Reply:
makes sense, also another point towards the custom parser. I would like `SHOW <OBJECT> [FROM <qualifier>] [LIKE <pattern>]` which is not supported in sql parser.